Documentation/Design/lambda_body_design.md

```diff
@@ -0,0 +1,72 @@
+# GLS Improvement: Lambda Support
+Engineer: faulda
```

Comment:
Would you be ok with phrasing this as a part of the language, not as an improvement doc? I'd like to keep these docs in after the PRs are done.